clients/src/main/java/org/apache/kafka/common/record/Records.java

```diff
@@ -18,32 +18,74 @@
 
 import java.io.IOException;
 import java.nio.channels.GatheringByteChannel;
+import java.util.Iterator;
 
 /**
- * A binary format which consists of a 4 byte size, an 8 byte offset, and the record bytes. See {@link MemoryRecords}
- * for the in-memory representation.
+ * Interface for accessing the records contained in a log. The log itself is represented as a sequence of log entries.
+ * Each log entry consists of a 4 byte size, an 8 byte offset, and a "shallow" {@link Record record}. If
```

Comment:
a 4 byte size, an 8 byte offset => an 8 byte offset, a 4 byte size of the record